test_log/src/main/java/org/corfudb/test_log/TestLog.java

```diff
@@ -0,0 +1,289 @@
+        // ...
+        fc_in.position(0);
+        long file_size = fc_in.size();
+        ByteBuffer md_buffer = ByteBuffer.allocate(MetadataSize);
+        int r = fc_in.read(md_buffer);
+        md_buffer.flip();
+        if (fc_out != null) {
+            fc_out.write(md_buffer);
+        }
+        if (r > 0) {
+            Metadata md = Metadata.parseFrom(md_buffer.array());
+            if (display) {
+                System.out.println("checksum " + String.format("%08x", md.getChecksum()));
+                System.out.println("length " + Integer.toString(md.getLength()));
+            }
+            ByteBuffer lh_buffer = ByteBuffer.allocate(LogHeaderSize);
+            r = fc_in.read(lh_buffer);
+            lh_buffer.flip();
+            if (fc_out != null) {
+                fc_out.write(lh_buffer);
+            }
+            if (r > 0) {
+                LogHeader lh = LogHeader.parseFrom(lh_buffer.array());
+                if (display) {
+                    System.out.println("version " + Integer.toString(lh.getVersion()));
+                    System.out.println("verify " + Boolean.toString(lh.getVerifyChecksum()));
+                }
+            }
+        }
+        HashFunction hf = Hashing.crc32c();
+        long rem_size = fc_in.size() - fc_in.position();  // if size == position then file pointer is off the end
+        int record_cnt = 0;
+        boolean replace_entry = false;
+        boolean enable_write = false;
+        switch (op.getOpType()) {
+            case ERASE_ONE:
+            case ERASE_REMAINING:
+                enable_write = true;
+                break;
+            default:
+                enable_write = false;
+        }
+        while (rem_size > 0) {
+            boolean trigger = false;
+            ByteBuffer comma_buffer = ByteBuffer.allocate(2);
+            int bytes_read = fc_in.read(comma_buffer);
+            comma_buffer.flip();
+            Short delim = comma_buffer.getShort();
+            comma_buffer.flip();
+            if (display) {
+                System.out.format("delimiter: %04x\n", delim);
+            }
+            if (delim != 0x4c45) {
+                System.out.println("Incorrect delimiter");
+            }
+            ByteBuffer md_buff = ByteBuffer.allocate(MetadataSize);
+            bytes_read += fc_in.read(md_buff);
+            md_buff.flip();
+            Metadata md = Metadata.parseFrom(md_buff.array());
+            if (display) {
+                System.out.format("Record length %d checksum %08x\n", md.getLength(), md.getChecksum());
+            }
+            ByteBuffer record_buffer = ByteBuffer.allocate(md.getLength());
+            bytes_read += fc_in.read(record_buffer);
+            record_buffer.flip();
+            LogEntry le = LogEntry.parseFrom(record_buffer.array());
+            HashCode hc = hf.newHasher()
+                    .putBytes(record_buffer.array())
+                    .hash();
+            if (hc.asInt() != md.getChecksum()) {
```

Comment:
what if checksum matches, do we print that fact ? Should these errors be on console only ?